dotCMS/src/main/java/com/dotcms/graphql/business/RelationshipFieldGenerator.java

```diff
@@ -0,0 +1,122 @@
+ */
+class RelationshipFieldGenerator implements GraphQLFieldGenerator {
+
+    private RelationshipAPI relationshipAPI = APILocator.getRelationshipAPI();
+
+    @VisibleForTesting
+    protected void setRelationshipAPI(RelationshipAPI relationshipAPI) {
+        this.relationshipAPI = relationshipAPI;
+    }
+
+    @Override
+    public GraphQLFieldDefinition generateField(final Field field) {
+        return createRelationshipField(field);
+    }
+
+    private GraphQLFieldDefinition createRelationshipField(
+            final Field field) {
+
+        final ContentType relatedContentType;
+        ContentType contentType = null;
+
+        try {
+            contentType = APILocator.getContentTypeAPI(APILocator.systemUser())
+                    .find(field.contentTypeId());
+            relatedContentType = getRelatedContentTypeForField(field);
+        } catch (DotSecurityException | DotDataException e) {
+            throw new FieldGenerationException(
+                    "Unable to create relationship field type for field: " + (contentType!=null ? contentType.variable() : "N/D")
+                            + "." + field.variable(), e);
+        }
+
+        Relationship relationship;
+
+        try {
+            relationship = relationshipAPI.getRelationshipFromField(field,
+                    APILocator.systemUser());
+        } catch (DotDataException | DotSecurityException e) {
+            throw new DotRuntimeException(e);
+        }
+
+        final ContentletRelationships contentletRelationships = new ContentletRelationships(null);
+        final ContentletRelationships.ContentletRelationshipRecords
+                records = contentletRelationships.new ContentletRelationshipRecords(
+                relationship,
+                relationshipAPI.isChildField(relationship, field));
+
+        GraphQLOutputType outputType = GraphQLTypeReference.typeRef(relatedContentType.variable());
+
+        outputType = records.doesAllowOnlyOne()
+                ? outputType
+                : list(outputType);
+
+        return newFieldDefinition()
+                .name(field.variable())
+                .type(field.required() ? nonNull(outputType) : outputType)
+                .dataFetcher(new RelationshipFieldDataFetcher()).build();
+    }
+
+    private ContentType getRelatedContentTypeForField(final Field field) {
+        DotPreconditions.checkNotNull(field, "Field can't be null");
+        DotPreconditions.checkArgument(UtilMethods.isSet(field.variable()), "Field variable needs to be set");
+        DotPreconditions.checkArgument(UtilMethods.isSet(field.relationType()), "Field relationType needs to be set");
+
+        final Relationship relationship;
+        try {
+            relationship = relationshipAPI.getRelationshipFromField(field,
+                    APILocator.systemUser());
+        } catch (DotSecurityException | DotDataException e) {
+            throw new FieldGenerationException("Relationship with name:"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)